src/util/getId.ts

```diff
@@ -0,0 +1,10 @@
+import { Compilation, Module } from 'webpack';
+import { Stats, Module as StatsModule } from '../types/Stats';
+
+export function getId(m: Module | StatsModule, stats: Stats | Compilation) {
+    if ('hooks' in stats) {
```

Comment:
>'hooks' in stats [](http://example.com/codeflow?start=8&length=16)

Might be worth adding an `isCompilation` utility and using that, rather than effectively doing feature detection at every step along the way.  What do you think?